### PR TITLE
Add DMIC pipelines to the CaVS HDA topology

### PR DESCRIPTION
**Pull request by ranj063**

### CODE REVIEW

File: tools/topology/topology2/cavs/platform/intel/tgl.conf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# TGL-specific variable definitions
```
Comment (plbossart) [inline tools/topology/topology2/cavs/platform/intel/tgl.conf:1]: also for ICL, JSL, ADL and follow-ups.

For simplicity we should make this the default and override it for APL-GLK with 19.2MHz and CNL/CML/CFL with 24MHz.

File: tools/topology/topology2/cavs/platform/intel/cnl.conf [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# CNL/CML/CFL-specific variable definitions
+Define {
+	DMIC_IO_CLK		24000000
+	DMIC_DRIVER_VERSION	1
```
Comment (plbossart) [inline tools/topology/topology2/cavs/platform/intel/cnl.conf:4]: DRIVER_VERSION?

Is this hardware version maybe? What does this refer to?

File: tools/topology/topology2/cavs/tplg.dot [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+"mixout.2.1"->"gain.2.1"
+
+"mixout.3.1"->"gain.3.1"
+
+"gain.3.1"->"copier.host.3.1"
+
+"gain.4.1"->"mixin.4.1"
+
+}
```
Comment (plbossart) [inline tools/topology/topology2/cavs/tplg.dot:32]: was this intended to be part of this PR?

File: tools/topology/topology2/cavs/cavs-passthrough-hdmi.conf [hunk 4]
```diff
@@ -19,13 +19,23 @@
...
+<dmic.conf>
+<pdm_config.conf>
 <hw_config.conf>
 <manifest.conf>
 <route.conf>
 <cavs/common_definitions.conf>
 
 Define {
 	HDA_CONFIG  "none"
+	PLATFORM "tgl"
```
Comment (plbossart) [inline tools/topology/topology2/cavs/cavs-passthrough-hdmi.conf:31]: I don't understand how this works for CNL, is this redefinting what is in CMakeList.txt

File: tools/topology/topology2/cavs/tplg.dot (hunk 3 above)
Comment (ranj063) [reply tools/topology/topology2/cavs/tplg.dot:32]: no. will remove

File: tools/topology/topology2/cavs/cavs-passthrough-hdmi.conf (hunk 4 above)
Comment (ranj063) [reply tools/topology/topology2/cavs/cavs-passthrough-hdmi.conf:31]: the default PLATFORM varaible definition is tgl, and is overridden by CMakelists.txt

File: tools/topology/topology2/cavs/platform/intel/cnl.conf (hunk 2 above)
Comment (ranj063) [reply tools/topology/topology2/cavs/platform/intel/cnl.conf:4]: @juimonen can you please comment?

Comment (lgirdwood) [reply tools/topology/topology2/cavs/platform/intel/cnl.conf:4]: The DMIC driver supports 3 versions of DMIC.

**Review by lgirdwood — COMMENTED** [review]
@singalsu @juimonen any comments here ?

File: tools/topology/topology2/cavs/platform/intel/cnl.conf (hunk 2 above)
Comment (juimonen) [reply tools/topology/topology2/cavs/platform/intel/cnl.conf:4]: yes dmic hw version

**Review by lgirdwood — APPROVED** [review]
Let's update the name to be DMIC_IP_VERSION since it's more meaningful. This can be done in another PR though.

**Pull request merged**